k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSetupCheckSettings.java

```diff
@@ -509,6 +516,43 @@ private void checkIncoming() throws MessagingException {
                     .synchronizeMailbox(account, account.getInboxFolder(), null, null);
         }
 
+        private void createSpecialLocalFolders(CheckDirection direction) throws MessagingException {
```

Comment:
this whole method seems like it doesn't really belong here. I'm not sure where it belongs though :hurtrealbad:

Reply:
I agree. I don't like any of the current account setup code. But I tried to keep the pull request focused to that one issue. This seemed like the least horrible place for the code without fundamentally changing how we do things.